WashU-IT-RIS/coldfront-wustl-fork
# ITDEV-34978 - Add Allocation Linkage

Adding a new class (plus migration) and associated code changes to handle parent/sub-allocations.

Testing:

1. Create an allocation as normal
2. Navigate to View All Allocations and confirm that it appears in the list with a `+` sign in the Sub-allocations column
3. Click the `+` sign and confirm that the form to create a sub-allocation appears
4. Modify any values you wish to and hit submit
5. Navigate back to View All Allocations and confirm that the new sub-allocation is listed underneath the original parent and that the row is greyed in background and does not have a '+' sign
6. Create a new parent allocation and then create a new sub-allocation for the *original* parent.  Confirm that the sub-allocation sorts with the original parent rather than the new parent.
7. Check and un-check the `No Grouping` selection in the filter form to show that the child allocations are sorted differently based on whether they are grouped with the parent or not.
